pydantic_extra_types/coordinate.py

```diff
@@ -29,37 +27,32 @@ def __get_pydantic_core_schema__(cls, source: type[Any], handler: GetCoreSchemaH
         return core_schema.float_schema(ge=cls.min, le=cls.max)
 
 
-@dataclass
+@dataclasses.dataclass
 class Coordinate(_repr.Representation):
-    _NULL_ISLAND: ClassVar[tuple[float, float]] = (0.0, 0.0)
+    _NULL_ISLAND: ClassVar[tuple[float, float]] = 0.0, 0.0
+
     latitude: Latitude
     longitude: Longitude
 
-    @classmethod
-    def __get_pydantic_json_schema__(cls, core_schema: CoreSchema, handler: GetJsonSchemaHandler) -> JsonSchemaValue:
-        field_schema: dict[str, Any] = handler(core_schema)
-        field_schema.update(format='coordinate')
-        return field_schema
-
     @classmethod
     def __get_pydantic_core_schema__(cls, source: type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        schema_chain = [
+            core_schema.no_info_wrap_validator_function(cls._parse_str, core_schema.str_schema()),
+            core_schema.no_info_wrap_validator_function(
+                cls._parse_tuple,
+                handler.generate_schema(Tuple[float, float]),
+            ),
+            handler(source),
+        ]
         return core_schema.no_info_wrap_validator_function(
             cls._parse_args,
-            core_schema.no_info_wrap_validator_function(
-                cls._parse_str,
-                core_schema.chain_schema(
-                    [
-                        core_schema.no_info_wrap_validator_function(
-                            cls._parse_tuple, handler.generate_schema(Tuple[Latitude, Longitude])
-                        ),
-                        handler(source),
-                    ]
-                ),
+            core_schema.union_schema(
+                [core_schema.chain_schema(schema_chain[2 - x :]) for x in range(3)],
```

Comment:
Thanks for your help, but I don't understand why `range(3)` and `schema_chain[2 - x :]` ?